src/Controls/src/Core/Platform/Android/TabbedPageManager.cs

```diff
@@ -655,76 +655,70 @@ protected virtual ColorStateList GetItemTextColorStates()
-				return null;
+			if (_orignalTabIconColors is null)
+				_orignalTabIconColors = IsBottomTabPlacement ? _bottomNavigationView.ItemIconTintList : _tabLayout.TabIconTint;
 
 			Color barItemColor = BarItemColor;
 			Color barSelectedItemColor = BarSelectedItemColor;
 
-			if (barItemColor == null && barSelectedItemColor == null)
+			if (barItemColor is null && barSelectedItemColor is null)
 				return _orignalTabIconColors;
 
-			if (_newTabIconColors != null)
+			if (_newTabIconColors is not null)
 				return _newTabIconColors;
 
 			int defaultColor;
+			int checkedColor;
 
 			if (barItemColor is not null)
-			{
 				defaultColor = barItemColor.ToPlatform().ToArgb();
-			}
 			else
-			{
+				defaultColor = GetDefaultColor();
+
+			if (barSelectedItemColor is not null)
+				checkedColor = barSelectedItemColor.ToPlatform().ToArgb();
+			else
+				checkedColor = GetDefaultColor();
+
+			_newTabIconColors = GetColorStateList(defaultColor, checkedColor);
+			return _newTabIconColors;
+		}
+
+		int GetDefaultColor()
+		{
+			int defaultColor;
 #pragma warning disable XAOBS001 // Obsolete
-				var styledAttributes =
-					TintTypedArray.ObtainStyledAttributes(_context.Context, null, Resource.Styleable.NavigationBarView, Resource.Attribute.bottomNavigationStyle, 0);
+			var styledAttributes =
+				TintTypedArray.ObtainStyledAttributes(_context.Context, null, Resource.Styleable.NavigationBarView, Resource.Attribute.bottomNavigationStyle, 0);
 #pragma warning restore XAOBS001 // Obsolete
 
-				try
+			try
+			{
+				var defaultColors = styledAttributes.GetColorStateList(Resource.Styleable.NavigationBarView_itemIconTint);
+				if (defaultColors is not null)
 				{
-					var defaultColors = styledAttributes.GetColorStateList(Resource.Styleable.NavigationBarView_itemIconTint);
-					if (defaultColors is not null)
-					{
-						defaultColor = defaultColors.DefaultColor;
-					}
-					else
-					{
-						// These are the defaults currently set inside android
-						// It's very unlikely we'll hit this path because the 
-						// NavigationBarView_itemIconTint should always resolve
-						// But just in case, we'll just hard code to some defaults
-						// instead of leaving the application in a broken state
-						if (IsDarkTheme)
-							defaultColor = new Color(1, 1, 1, 0.6f).ToPlatform();
-						else
-							defaultColor = new Color(0, 0, 0, 0.6f).ToPlatform();
-					}
+					defaultColor = defaultColors.DefaultColor;
 				}
-				finally
+				else
 				{
-					styledAttributes.Recycle();
+					// These are the defaults currently set inside android
+					// It's very unlikely we'll hit this path because the 
+					// NavigationBarView_itemIconTint should always resolve
+					// But just in case, we'll just hard code to some defaults
+					// instead of leaving the application in a broken state
+					if (IsDarkTheme)
+						defaultColor = new Color(1, 1, 1, 0.6f).ToPlatform();
```

Comment:
Is possible to use a resource system color?

Reply:
OR point to where in android we grabbed these default values.

Reply:
- The use of `hardcoded` colors has already been implemented as a fallback. If retrieving the default color from style attributes fails, the application avoids breaking by assigning a predefined default color. 

- I have now updated the code to fetch the resource color and assign it to the default color based on the theme. Please let me know if you have any other concerns. @jfversluis / @rmarinho